### PR TITLE
fix[ux]: error messages relating to initializer issues

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
fix some error messages

### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/semantics/analysis/module.py [hunk 1]
```diff
@@ -489,7 +495,22 @@ def visit_InitializesDecl(self, node):
...
+            rhs = None
+            # find the alias of the uninitialized module in this contract
+            # to fill out the error message with.
+            for k, v in self.namespace.items():
+                if isinstance(v, ModuleInfo) and v.module_t == item.module_t:
+                    rhs = k
+                    break
+
+            if rhs is None:
+                hint = "try importing {item.alias} first"
```
Comment (pcaversaccio) [inline vyper/semantics/analysis/module.py:509]: Currently untested.

File: vyper/semantics/analysis/module.py [hunk 2]
```diff
@@ -489,7 +495,22 @@ def visit_InitializesDecl(self, node):
...
+            elif isinstance(module_ref, vy_ast.Name):
+                hint = f"did you mean {module_ref.id}[{lhs} := {rhs}]?"
+            else:
+                hint = f"add `{lhs} := {rhs}` to its initializer list"
```
Comment (pcaversaccio) [inline vyper/semantics/analysis/module.py:513]: Currently untested.

**Review by pcaversaccio — APPROVED** [review]

**Review by fubuloubu — COMMENTED** [review]
Kind of a code smell that only one test changed with all of these updates

**Comment by cyberthirst** [comment]
consider:
main
```python
import lib1
import lib2
import lib3

initializes: lib2[lib1:=lib1]
initializes: lib1
initializes: lib3
```
lib1
```python
counter: uint256
```
lib3
```python
counter: uint256
```
lib2
```python
import lib1
import lib3

uses: lib1
uses: lib3

counter: uint256

@internal
def foo():
    lib1.counter += 1
    lib3.counter += 1
```
then:
```
vyper.exceptions.InitializerException: `lib2` uses `lib3`, but it is not initialized with `lib3`

  (hint: did you mean lib2[lib3 := lib3]?)

  contract "tests/custom/test4.vy:5", line 5:0 
       4
  ---> 5 initializes: lib2[lib1:=lib1]
  -------^
       6 initializes: lib1
```
ie the hint excludes the current `lib1` intializer - `initializes: lib2[lib1:=lib1]`

File: vyper/semantics/analysis/module.py (hunk 1 above)
Comment (cyberthirst) [inline vyper/semantics/analysis/module.py:509]: this should be an f-string, right

**Comment by charles-cooper** [comment]
note: planning on changing the initializer syntax to look like `lib[dep]` instead of `lib[dep := dep]`

**Comment by charles-cooper** [comment]
> Kind of a code smell that only one test changed with all of these updates

it was missing tests for the new cases -- thanks @cyberthirst for help adding

**Comment by charles-cooper** [comment]
> note: planning on changing the initializer syntax to look like `lib[dep]` instead of `lib[dep := dep]`

after thinking about it i think this is kind of a non-issue, both syntaxes are kind of ok, because the latter syntax helps make it clear when you use `lib.dep` and the library imports it as a different name (`e.g., lib[dependency := dep]`). in this case if you access `dep` through the library (as `lib.dependency`) it's clearer that `dep` and `lib.dependency` are the same module. deciding on one syntax over the other is splitting hairs IMO. (there COULD be an argument that we should ban accessing `dep` through `lib.dependency` altogether. but i don't think it's a good time to revisit this so close to release, we can always revisit in a later release).

**Review by cyberthirst — APPROVED** [review]

**Comment by pcaversaccio** [comment]
> > note: planning on changing the initializer syntax to look like `lib[dep]` instead of `lib[dep := dep]`
> 
> after thinking about it i think this is kind of a non-issue, both syntaxes are kind of ok, because the latter syntax helps make it clear when you use `lib.dep` and the library imports it as a different name (`e.g., lib[dependency := dep]`). in this case if you access `dep` through the library (as `lib.dependency`) it's clearer that `dep` and `lib.dependency` are the same module. deciding on one syntax over the other is splitting hairs IMO. (there COULD be an argument that we should ban accessing `dep` through `lib.dependency` altogether. but i don't think it's a good time to revisit this so close to release, we can always revisit in a later release).

FWIW, I use the `lib[dep := dep]` syntax in snekmate contract mocks; [example](https://github.com/pcaversaccio/snekmate/blob/dba2f276b310d8b32067316c84b3c9fa1e324ffe/src/snekmate/auth/mocks/Ownable2StepMock.vy#L17):

```vy
# @dev We import and initialise the `Ownable` module.
from .. import Ownable as ow
initializes: ow


# @dev We import and initialise the `Ownable2Step` module.
from .. import Ownable2Step as o2
initializes: o2[ownable := ow]
```

**Pull request merged**